### DISCUSSION TITLE
new repo request: nixd, the nix language server based on official nix library

### DISCUSSION THREAD

**Discussion by inclyc** [nix-community/infra#611] [Q&A] [answered]
I wanted to share a project I've been working on recently. It's a nix language server that's based on NixOS/nix directly (i.e. is linked with), and it's designed to improve **consistency** between nix commands and the editor experience. While it's still a work in progress, I've already implemented a few key features such as `hover`, `workspace/configuration`, `completion`, and basic diagnostics. I was thinking of putting this project in the nix-community incubator to get more visibility. Would you be interested in trying it out and providing feedback? Thanks!

Project link: https://github.com/inclyc/nixd

Some notable features provided by linking with the Nix library include:

- Diagnostics and evaluation that produce identical results as the real Nix command.
- Shared eval caches (flake, file) with your system's Nix.
- Native support for cross-file analysis.
- Precise Nix language support. We do not maintain "yet another parser & evaluator".
- Support for built-ins, including Nix plugins.

**Comment by inclyc** [comment]
Project matrix room: https://matrix.to/#/#nixd:matrix.org

**Comment by adisbladis** [comment]
I have invited @inclyc to the nix-community org and explained that the repo needs to be transferred.

**Comment by inclyc** [comment, marked as answer]
Thanks! Migrated here: https://github.com/nix-community/nixd
